react/src/components/BoxReconciliationOverlay/BoxReconciliationOverlay.tsx

```diff
@@ -0,0 +1,214 @@
+/* eslint-disable no-console */
+import { useCallback, useContext, useEffect, useMemo } from "react";
+import { useMutation, useQuery, useReactiveVar } from "@apollo/client";
+import { boxReconciliationOverlayVar } from "queries/cache";
+import { useErrorHandling } from "hooks/useErrorHandling";
+import { useNotification } from "hooks/useNotification";
+import {
+  BoxState,
+  ShipmentByIdWithProductsAndLocationsQuery,
+  ShipmentByIdWithProductsAndLocationsQueryVariables,
+  ShipmentDetail,
+  UpdateShipmentWhenReceivingMutation,
+  UpdateShipmentWhenReceivingMutationVariables,
+} from "types/generated/graphql";
+import { GlobalPreferencesContext } from "providers/GlobalPreferencesProvider";
+import { SHIPMENT_BY_ID_WITH_PRODUCTS_AND_LOCATIONS_QUERY } from "queries/queries";
+import { UPDATE_SHIPMENT_WHEN_RECEIVING } from "queries/mutations";
+import {
+  BoxReconciliationView,
+  ILocationData,
+  IProductWithSizeRangeData,
+} from "./components/BoxReconciliationView";
+
+export interface IBoxReconciliationOverlayData {
+  shipmentDetail: ShipmentDetail;
+}
+
+export function BoxReconciliationOverlay() {
+  const { createToast } = useNotification();
+  const { triggerError } = useErrorHandling();
+  const { globalPreferences } = useContext(GlobalPreferencesContext);
+  const baseId = globalPreferences.selectedBaseId;
+  const boxReconciliationOverlayState = useReactiveVar(boxReconciliationOverlayVar);
+
+  const onOverlayClose = useCallback(() => {
+    boxReconciliationOverlayVar({
+      isOpen: false,
+      labelIdentifier: undefined,
+      shipmentId: undefined,
+    });
+  }, []);
+
+  const { loading, error, data } = useQuery<
+    ShipmentByIdWithProductsAndLocationsQuery,
+    ShipmentByIdWithProductsAndLocationsQueryVariables
+  >(SHIPMENT_BY_ID_WITH_PRODUCTS_AND_LOCATIONS_QUERY, {
+    variables: {
+      shipmentId: boxReconciliationOverlayState.shipmentId || "",
+      baseId: baseId || "",
+    },
+    skip: !boxReconciliationOverlayState.shipmentId || !baseId,
+  });
+
+  const [updateShipmentWhenReceiving, updateShipmentWhenReceivingStatus] = useMutation<
```

Comment:
@vahidbazzaz In this mutation u either need to update the cache after the mutation or run a refetchquery. Otherwise, the list and counter in ReceiveShipment does not update.

Reply:
@HaGuesto Since it already contains the required information, it does not need to be changed